ui/raidboss/emulator/data/Encounter.ts

```diff
@@ -56,14 +56,14 @@ export default class Encounter {
   initialize(): void {
     const startStatuses = new Set<string>();
 
-    this.logLines.forEach((line, i) => {
+    for (const line of this.logLines) {
       if (!line)
         throw new UnreachableCode();
 
       let res: MatchStartInfo | MatchEndInfo | undefined =
           EmulatorCommon.matchStart(line.networkLine);
       if (res) {
-        this.firstLineIndex = i;
+        this.firstLineIndex = this.logLines.indexOf(line);
```

Comment:
This is premature optimation, but surely keeping some `lineIdx` variable on the side is cheaper than looking through all the log lines to find this line?

Reply:
Lines already track an index but it's indexed against the original log file, not the encounter, which doesn't really make sense. Maybe that should be re-indexed after finalizing the encounter?

Reply:
Sorry, I'm not sure I understand your suggestion.  I was more just suggesting a micro-optimization here.  `indexOf` is a walk through all of the log lines vs the previous code which just used an incremented variable.  If you don't think that's useful, then ignore this suggestion.

Reply:
During log file parsing during import, the process is basically:

1. Convert log file to `LineEvent`s
2. Split `LineEvent`s into `Encounter`s

During the convert portion this code runs, indexing the `LineEvent`s:
https://github.com/quisquous/cactbot/blob/6a6688cb74b15e9418119e27cfcb380085cb4311/ui/raidboss/emulator/data/NetworkLogConverter.ts#L24-L27

This indexes lines based on their position in the original log file. It should probably be moved entirely, and index them based on their position in the encounter instead, e.g. just before creating the `new Encounter` here:
https://github.com/quisquous/cactbot/blob/6a6688cb74b15e9418119e27cfcb380085cb4311/ui/raidboss/emulator/data/NetworkLogConverter.worker.ts#L27-L29